### PR TITLE
[HttpKernel] rename ExceptionEvent to ErrorEvent and remove the ErrorException wrapper

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | -
| License       | MIT
| Doc PR        | -

Replaces #34232

This PR gets rid of the `FatalThrowableError` wrapper we introduced when doing the transition from PHP 5 to PHP 7 - to handle all `Throwable` in a seamless way.

It does so by replacing the `kernel.exception` event by the `kernel.error` one. The new name is consistent with the `console.error` event that we introduced for similar reasons in the Console component.

Along the way, some method/classes are renamed.
The `ErrorException` class from the `ErrorHandler` component is removed.

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/HttpKernel.php [hunk 1]
```diff
@@ -203,8 +205,29 @@ private function finishRequest(Request $request, int $type)
      */
     private function handleThrowable(\Throwable $e, Request $request, int $type): Response
     {
-        $event = new ExceptionEvent($this, $request, $type, $e);
-        $this->dispatcher->dispatch($event, KernelEvents::EXCEPTION);
+        if (!$this->dispatcher instanceof \Symfony\Component\EventDispatcher\EventDispatcherInterface) {
+            @trigger_error('Listening to the "kernel.exception" event is deprecated since Symfony 4.4, listen to "kernel.error" instead.', E_USER_DEPRECATED);
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/HttpKernel.php]: how can affected people get rid of the deprecation?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: I think we don't have any other alternative here: ppl that hit this code path must bump to v5.
Good news, this affects ppl that implement `EventDispatcherInterface` from the contracts ONLY. If such an implementation exists (does it), it must be very new, and very rare.

**Comment by nicolas-grekas** [comment]
PR is ready.

**Comment by Tobion** [comment]
Fine for me. But I don't get why this would be better than https://github.com/symfony/symfony/pull/33065#issuecomment-549446194.
This new solution just achieves the same thing but forces a much bigger bc break.

**Comment by nicolas-grekas** [comment]
>  This new solution just achieves the same thing but forces a much bigger bc break and many more affected users.

In terms of the amount of code, renaming the methods on the `ExceptionEvent` class has a similar impact as renaming the class to `ErrorEvent` (without renaming the methods). Then, this also requires ppl to change the way their classes are registered. We could consider this additional requirement a positive one, as it makes the intent more clear, by requiring an explicit opt-in for the new sub-API.

It's also a matter of target names - which one makes more sense: `ExceptionEvent::get/setThrowable()` or `ErrorEvent::get/setException()`?

My personal preference goes for the latter, also because it is consistent with the `console.error` and the corresponding `ConsoleErrorEvent`.

**Comment by nicolas-grekas** [comment]
Smaller diff, we're going to prefer #34309

**Pull request closed without merging**